src/navigation/HardwareWalletTxNavigator.tsx

```diff
@@ -1,18 +1,111 @@
 import { createMaterialTopTabNavigator } from '@react-navigation/material-top-tabs';
-import React from 'react';
+import React, { useCallback, useEffect } from 'react';
 import Routes from '@/navigation/routesNames';
 import { BackgroundProvider } from '@/design-system';
-import { useDimensions } from '@/hooks';
+import { useDimensions, useWallets } from '@/hooks';
 import { PairHardwareWalletAgainSheet } from '@/screens/hardware-wallets/PairHardwareWalletAgainSheet';
 import { PairHardwareWalletErrorSheet } from '@/screens/hardware-wallets/PairHardwareWalletErrorSheet';
 import { SimpleSheet } from '@/components/sheet/SimpleSheet';
+import { useLedgerConnect } from '@/hooks/useLedgerConnect';
+import { LEDGER_ERROR_CODES } from '@/utils/ledger';
+import { useNavigation } from '@/navigation';
+import { logger } from '@/logger';
+import { DebugContext } from '@/logger/debugContext';
+// eslint-disable-next-line no-restricted-imports
+import { RouteProp, useRoute } from '@react-navigation/core';
+import { atom, useRecoilState } from 'recoil';
+import { MMKV } from 'react-native-mmkv';
+
+export const ledgerStorage = new MMKV({
+  id: 'ledgerStorage',
+});
+
+export const HARDWARE_TX_ERROR_KEY = 'hardwareTXError';
+
+export const setHardwareTXError = (value: boolean) => {
+  logger.info(`setHardwareTXError`, { value });
+  ledgerStorage.set(HARDWARE_TX_ERROR_KEY, value);
+};
 
 const Swipe = createMaterialTopTabNavigator();
 export const HARDWARE_WALLET_TX_NAVIGATOR_SHEET_HEIGHT = 534;
 
+export type HardwareWalletTxParams = {
+  submit: () => void;
+};
+
+type RouteParams = {
+  HardwareWalletTxParams: HardwareWalletTxParams;
+};
+
+// atoms used for navigator state
+export const LedgerIsReadyAtom = atom({
```

Comment:
happy to use a diff lib here just using what we have in the codebase without using redux

Reply:
I think that's okay, I made a directory for storing recoil stuff like atoms at `src/recoil` some time ago, you could move it if you want